lib/opensusebasetest.pm

```diff
@@ -638,6 +638,52 @@ sub wait_grub_to_boot_on_local_disk {
     }
 }
 
+sub grub_select {
+    if ((my $grub_nondefault = get_var('GRUB_BOOT_NONDEFAULT', 0)) gt 0) {
+        my $menu = $grub_nondefault * 2 + 1;
+        bmwqemu::fctinfo("Boot non-default grub option $grub_nondefault (menu item $menu)");
+        boot_grub_item($menu);
+    } elsif (my $first_menu = get_var('GRUB_SELECT_FIRST_MENU')) {
+        if (my $second_menu = get_var('GRUB_SELECT_SECOND_MENU')) {
+            bmwqemu::fctinfo("Boot $first_menu > $second_menu");
+            boot_grub_item($first_menu, $second_menu);
+        } else {
+            bmwqemu::fctinfo("Boot $first_menu");
+            boot_grub_item($first_menu);
+        }
+    }
+    elsif (!get_var('S390_ZKVM')) {
+        # confirm default choice
+        send_key 'ret';
+    }
+}
+
+sub handle_grub {
+    my ($self, %args) = @_;
+    my $bootloader_time  = $args{bootloader_time};
+    my $in_grub          = $args{in_grub};
+    my $linux_boot_entry = $args{linux_boot_entry} // (is_sle('15+') ? 15 : 14);
+
+    # On Xen PV and svirt we don't see a Grub menu
```

Comment:
Only a minor nit, but you are copying a block of text which appears to be unchanged. This hides the change history, so you will get the blame for problems this code has.

Reply:
This one fixed, verification http://quasar.suse.cz/tests/4674